lntest/harness_net.go

```diff
@@ -962,23 +962,19 @@ func saveProfilesPage(node *HarnessNode) error {
 // waitForTxInMempool blocks until the target txid is seen in the mempool. If
 // the transaction isn't seen within the network before the passed timeout,
 // then an error is returned.
-func (n *NetworkHarness) waitForTxInMempool(ctx context.Context,
-	txid chainhash.Hash) error {
-
-	// Return immediately if harness has been torn down.
-	select {
-	case <-n.runCtx.Done():
-		return fmt.Errorf("NetworkHarness has been torn down")
-	default:
-	}
-
+func (n *NetworkHarness) waitForTxInMempool(txid chainhash.Hash) error {
 	ticker := time.NewTicker(50 * time.Millisecond)
 	defer ticker.Stop()
 
+	ctxt, cancel := context.WithTimeout(n.runCtx, DefaultTimeout)
+	defer cancel()
+
 	var mempool []*chainhash.Hash
 	for {
 		select {
```

Comment:
In this select, `ctxt` is a wrapper of `n.runCtx`. If the parent gets cancelled, `<-n.runCtx.Done()` and `<-ctxt.Done()` will get a message. In Go, if more than one `case` of a select are ready the runtime picks one at random. This means that if `n,runCtx` gets cancel our logs may show some wrong messages about `wanted %v, found %v txs in mempool: %v . 

Here it is not critical, but better wrap background than runCtx.

Reply:
Very good point! The reason I want to use `runCtx` instead of background context is to provide a better structure over the test components, such that a top parent context can easily cancel all its descendants, which will be much clearer in later PRs.

As for the multiple `context.Done`, they are all fixed in later PRs, in which we use a timer instead of a timeout context.